Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/exceptions/KnownExceptions.kt

```diff
@@ -0,0 +1,29 @@
+package de.rki.coronawarnapp.bugreporting.exceptions
+
+import android.content.Context
+import com.google.android.gms.common.api.ApiException
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.util.HumanReadableError
+
+interface KnownException {
+    fun matches(throwable: Throwable): Boolean
+
+    fun getHRE(context: Context): HumanReadableError
+}
+
+enum class KnownExceptions : KnownException {
+    // ApiException with StatusCode 17
+    ENS_NOT_INSTALLED {
+        override fun matches(throwable: Throwable): Boolean = throwable is ApiException && throwable.statusCode == 17
+
+        override fun getHRE(context: Context): HumanReadableError = HumanReadableError(
+            title = "${context.getString(R.string.errors_generic_details_headline)}: 3\n" +
+                context.getString(R.string.errors_generic_headline),
+            description = context.getString(R.string.errors_google_update_needed)
+        )
+    },
+}
```

Comment:
My idea is for future updates when we have more time:
* Expand this, consume all cases 
* Replace `ExceptionReporter.kt` with a more central flow that can be subscribed too, hooking somewhere into the error classes by @BMItter 
* Subscribe to this errorFlow in MainActivity
* Remove the whole "application accesses activity lifecycle to setup `ErrorReportReceiver` , then remove the whole `ErrorReportReceiver`
* Now we have a single flow that supplies error data, show popups for now, in the future future, replace popup showing with a homescreen card for errors.